### PR TITLE
Adding Doxygen headers to cvtFast and freeList; Correcting a markdown link in release_notes

### PR DESCRIPTION
**Pull request by KathrynBaker**
Apologies that it is three files, rather than one, hopefully, they have an appropriate level of detail.

### CODE REVIEW

File: modules/libcom/src/freeList/freeList.h [hunk 1]
```diff
@@ -7,27 +7,38 @@
...
+ *
+ * \brief Allocate and free fixed size memory elements.
+ *
+ * \details
+ * Describes routines to allocate and free fixed size memory elements.
+ * Free elements are maintained on a free list rather than being returned to the heap via calls to free.
+ * When it is necessary to call malloc(), memory is allocated in multiples of the element size.
+ */
+ 
+ #ifndef INCfreeListh
```
Comment (mdavidsaver) [inline modules/libcom/src/freeList/freeList.h]: It looks like everything after the leading comment block has been unintentionally indented by one space.

File: documentation/RELEASE_NOTES.md [hunk 2]
```diff
@@ -457,7 +457,7 @@ work with older Base releases.
 
 This would also be a good time to modify the device support to use the type-safe
 device support entry tables that were introduced in Base-3.16.2 -- see
-[#type-safe-device-and-driver-support-tables](this entry below) for the
+[this entry below](#type-safe-device-and-driver-support-tables) for the
```
Comment (anjohnson) [inline documentation/RELEASE_NOTES.md:460]: Oops, thanks for catching this! I may commit it separately.

File: modules/libcom/src/cvtFast/cvtFast.h [hunk 3]
```diff
@@ -27,9 +30,7 @@
 extern "C" {
 #endif
 
-/*
- * All functions return the number of characters in the output
- */
+/** \brief All functions return the number of characters in the output */
```
Comment (anjohnson) [inline modules/libcom/src/cvtFast/cvtFast.h]: This `\brief` will only apply the comment to the first function below it. If you're not going to add a description of each one with a `\returns` line I suggest moving the comment into the file description at the top, preferably as a separate paragraph.

**Review by anjohnson — COMMENTED** [review]
A couple of comments, and as @mdavidsaver requested please undo the code indentation in freeList.h.

Thanks.

**Comment by mdavidsaver** [comment]
Applied as dd09afc4e86ba96521a38af0944709b0c5fa1ff4.  Thanks!

**Pull request closed without merging**